nano/node/blockprocessor.cpp

```diff
@@ -156,7 +156,18 @@ void nano::block_processor::verify_state_blocks (nano::unique_lock<std::mutex> &
 	assert (!mutex.try_lock ());
 	nano::timer<std::chrono::milliseconds> timer_l (nano::timer_state::started);
 	std::deque<nano::unchecked_info> items;
-	items.swap (state_blocks);
+	if (state_blocks.size () <= max_count)
+	{
+		items.swap (state_blocks);
+	}
+	else
+	{
+		for (auto i (0); i < max_count && !state_blocks.empty (); ++i)
```

Comment:
!state_blocks.empty() should be false due to the if condition, right? An assert instead inside the loop?

Reply:
Should be, yes